zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/PerServerConnectionPool.java

```diff
@@ -398,4 +423,26 @@ private static InetAddress getSelectedHostString(SocketAddress addr) {
         }
     }
 
+    /**
+     * Closes idle connections in the connection pool for a given EventLoop. The closing is performed on the EventLoop
+     * thread since the connection pool is not thread safe.
+     *
+     * @param eventLoop - the event loop to drain
+     */
+    void drainIdleConnectionsOnEventLoop(EventLoop eventLoop) {
+        eventLoop.execute(() -> {
+            Deque<PooledConnection> connections = connectionsPerEventLoop.get(eventLoop);
+            for(Iterator<PooledConnection> it = connections.iterator(); it.hasNext(); ) {
```

Comment:
nit: Any reason to use a iterator here, vs a for-each? 
For this case, it should  just use a iterator underneath, if performance vs readability is the tradeoff.

Reply:
I used an iterator so I could use remove(). However, I don't need to be removing the connection from the backing Deque (it was required on an earlier implementation I had). I can switch to a for-each